### DISCUSSION TITLE
Problem building the development branch

### DISCUSSION THREAD

**Discussion by jostle** [Audiveris/audiveris#727] [General]
I am trying to build the development branch but getting an error
+
`~/src/audiveris.development/audiveris$ ./gradlew build`

> Configure project :
minJavaVersion: 21
hostOS: linux-x86_64

> Configure project :schemas
Xslt transformer: XsltProc

> Task :compileJava FAILED

`FAILURE: Build failed with an exception.`

* What went wrong:
Execution failed for task ':compileJava'.
> error: invalid source release: 21

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 752ms
3 actionable tasks: 3 executed
~/src/audiveris.development/audiveris$`

I have checked that my default java version is 21.0.2
`~/src/audiveris.development/audiveris$ javac -version`
`javac 21.0.2`

Is there somewhere in the gradle configuration that overrides my default javac version?

BTW this is a clean 
`git clone https://github.com/Audiveris/audiveris.git`
`git checkout development`
`git pull --all`
`./gradlew build`

**Comment by hbitteur** [comment]
> Is there somewhere in the gradle configuration that overrides my default javac version?

Not to my knowledge.

Java 21 is indeed the version required for the development branch.
This should be OK with your environment.
Are you sure you don't have several Java environments installed, and you are just picking up the wrong one?
Also, check JAVA_HOME environment variable.

**Comment by jostle** [comment]
Solved ...
I had, at some stage in the past, added this line to my `~/.gradle/gradle.properties` file :
`org.gradle.java.home=/usr/lib/jvm/java-17-openjdk-amd64`

Removing that line I can build and run.

Thanks

**Comment by jostle** [comment]
Closing
